### ISSUE TITLE
MD044: Names set in 'names' configuration are flagged even when in backticks in YAML frontmatter

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#513]
### rumdl version

0.1.44

### Installation method

mise

### Bug description

Rule MD044 (https://github.com/rvben/rumdl/blob/950b3b6d8217175b1611eae5a89c4fdad99e43e2/docs/md044.md) ensures proper capitalisation of proper nouns.

Sometimes command-line tooling will be called an all-lowercase version of the proper noun. Usually, the name of command-line tooling like this is placed in `backticks` so that it's acceptable despite `names` configuration in rule MD044.

However, placing the name of command-lint tooling like this in `backticks` in frontmatter is still flagged as a violation when using `title` in YAML frontmatter as a level 1 heading. This is true even though placing the same level 1 heading in regular `# heading` syntax works fine.

### Steps to reproduce

1. Create a Markdown file with the following content:
   ```markdown
   ---
   title: `goodapplication` CLI
   ---
   
   Introductory `goodapplication` CLI text.
   ```
1. Create a `.rumdl.toml` file with the following content:
   ```toml
   [MD044]
   names = ["GoodApplication"]
   ```
1. Run `rumdl check` on the Markdown.

### Expected behavior

Given the scenario above, I'd expect no violations.

### Actual behavior

Given the scenario above, I get:

```shell
test.md:2:9: [MD044] Proper name 'goodapplication' should be 'GoodApplication' [*]
```

### Configuration

```toml
[MD044]
names = ["GoodApplication"]
```

### Operating system

macOS

### Additional context

The following Markdown is acceptable to rule MD044:

```markdown
# `goodapplication` CLI

Introductory `goodapplication` CLI text.
```

**Comment by rvben** [comment]
Fixed in v0.1.45. Names inside backtick code spans in YAML frontmatter are no longer flagged when `code_blocks = false`. Thanks for reporting @eread!
